picasso/src/test/java/com/squareup/picasso/BitmapHunterTest.java

```diff
@@ -637,6 +637,31 @@
 
     assertThat(shadowMatrix.getPreOperations()).containsOnly("scale 0.5 0.5");
   }
+  
+  @Test public void onlyScaleDownOriginalBigger() throws Exception {
+    Bitmap source = Bitmap.createBitmap(100, 100, ARGB_8888);
+    Request data = new Request.Builder(URI_1).resize(50, 50).onlyScaleDown().build();
+    Bitmap result = transformResult(data, source, 0);
+    
+    ShadowBitmap shadowBitmap = shadowOf(result);
+    assertThat(shadowBitmap.getCreatedFromBitmap()).isSameAs(source);
```

Comment:
This is inside onlyScaleDownOriginalBigger(), result and source shouldnt be the same. You probably meant the code in line 659